### PR TITLE
feat: Jenkins repo support private repo

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description
- `Jenkins` support clone private repo
- delete `scm.projectRepo` and `scm.projectBranch` field
- add `scm.cloneURL` for clone config, `scm.apiURL` for api request, `scm.branch` for repo branch config
- add `GITLAB_SSHKEY` environment variable for private repo

## Related Issues
#993 

## New Behavior (screenshots if needed)
- create `Jenkins` credential for private key
image
- create job with this credential
image

### CODE REVIEW

File: docs/plugins/jenkins-pipeline.zh.md [hunk 1]
```diff
@@ -35,6 +35,7 @@
 ```shell
 export IMAGE_REPO_PASSWORD=YOUR_IMAGE_REPO_PASSWORD
 export GITLAB_TOKEN=YOUR_GITLAB_TOKEN
+export GITLAB_SSHKEY=YOUR_REPO_PRIVATE_KEY
```
Comment (daniel-hutao) [inline docs/plugins/jenkins-pipeline.zh.md:38]: What does YOUR_REPO_PRIVATE_KEY mean?

Comment (daniel-hutao) [reply docs/plugins/jenkins-pipeline.zh.md:38]: > This is your ssh private key for clone repo

Oh, yeah I've got it. I thought it's token yesterday.

File: docs/plugins/jenkins-pipeline.zh.md [hunk 2]
```diff
@@ -64,8 +65,9 @@ tools:
       user: admin
       enableRestart: true
     scm:
-      projectURL: YOUR_JENKINS_ADDR/root/spring-demo
-      projectBranch: master
+      cloneURL: git@YOUR_REPO_CLONE_ADDRESS/root/spring-demo
```
Comment (daniel-hutao) [inline docs/plugins/jenkins-pipeline.zh.md:68]: Please help to update `docs/best-practices/gitlab-jenkins-harbor.zh.md` as well.

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

File: internal/pkg/plugininstaller/jenkins/option.go [hunk 3]
```diff
@@ -210,3 +219,13 @@ func (j *JobOptions) buildCIConfig() {
 	}
 	j.CIConfig = ciConfig
 }
+
+func (j *JobOptions) createGitlabSSHPrivateKey(jenkinsClient jenkins.JenkinsAPI) error {
+	if j.SCM.SSHprivateKey == "" {
+		log.Warnf("jenkins gittlab ssh key not config, private repo can't be clone")
+		return nil
```
Comment (aFlyBird0) [inline internal/pkg/plugininstaller/jenkins/option.go:226]: why should here `return nil`?

**Comment by daniel-hutao** [comment]
Remember close #1102 after this pr merged.

**Review by daniel-hutao — APPROVED** [review]

**Review by aFlyBird0 — APPROVED** [review]

**Pull request merged**